src/Xml/Writer/ElementValueBuilder.php

```diff
@@ -49,12 +49,18 @@ private function buildXsiType(XMLWriter $writer): Generator
 
         $context = $this->context;
         $type = $context->type;
+        $xsiType = $this->encoder instanceof Feature\XsiTypeCalculator
+            ? $this->encoder->resolveXsiTypeForValue($context, $this->value)
+            : XsiTypeDetector::detectFromValue($context, $this->value);
+        $includeXsiTargetNamespace = $this->encoder instanceof Feature\XsiTypeCalculator
```

Comment:
I'm not completely sure about this : one might want to use the 'default' logic even from the custom XsiTypeCalculator.  Maybe the default logic should be moved to something like `XsiTypeDetector::shouldIncludeXsiTargetNamespace()` so that it can be reused from the encoder. However, for the ObjectEncoder, different rules apply.

Maybe if one cares to change the xsi:type manually, they also care about deciding wether that namespace is imported?